discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -60,10 +129,42 @@ pub(crate) async fn handle_user_interaction(
     }
 }
 
+/// Computes [`PositiveCoi`]s weights used to determine how many documents to fetch using each centers' embedding.
+fn compute_coi_weights(cois: &[PositiveCoi], horizon: Duration) -> Vec<f32> {
+    let relevances = compute_coi_relevances(cois, horizon, system_time_now());
+    let relevances = relevances
+        .iter()
+        .map(|rel| 1.0 - (-3.0 * rel).exp())
+        .collect_vec();
+
+    let rel_sum: f32 = relevances.iter().sum();
+    relevances.iter().map(|rel| rel / rel_sum).collect()
```

Comment:
🔧 we have to guard against division by zero here, as all relevances and hence their sum could be zero.